src/Provider/AbstractProvider.php

```diff
@@ -619,8 +616,8 @@ public function getParsedResponse(RequestInterface $request)
     {
         try {
             $response = $this->getResponse($request);
-        } catch (BadResponseException $e) {
-            $response = $e->getResponse();
+        } catch (ClientExceptionInterface $e) {
+            throw $e; 
```

Comment:
There's no equivalent here.

Reply:
True, But that is a good thing =)

The ClientException will only be thrown when it is impossible to generate a response. Ie network errors or the Request is invalid (missing method) etc.

Reply:
OK, so we can check for either of those and try to construct a client.